apps/opik-backend/src/main/java/com/comet/opik/api/resources/v1/priv/LlmProviderApiKeyResource.java

```diff
@@ -123,4 +124,20 @@ public Response updateApiKey(@PathParam("id") UUID id,
 
         return Response.noContent().build();
     }
+
+    @DELETE
+    @Path("{id}")
+    @Operation(operationId = "deleteLlmProviderApiKey", summary = "Delete LLM Provider's ApiKey by id", description = "Delete LLM Provider's ApiKey by id", responses = {
+            @ApiResponse(responseCode = "204", description = "No Content"),
+            @ApiResponse(responseCode = "404", description = "Not found", content = @Content(schema = @Schema(implementation = ErrorMessage.class)))
+    })
+    public Response deleteById(@PathParam("id") UUID id) {
```

Comment:
We currently offer a batch deletion endpoint for the majority of our resources. I encourage you to implement one, as it's likely going to be required by the FE soon e.g: a user checks multiple keys for deletion.

If you do it now, I honestly think there's no reason to keep both the batch and no-batch version. It'b be less maintenance to just offer the batch version.

Anyway, this is not a blocker.

Reply:
I added a batch deletion, but also left delete by Id. All other resources have such endpoint and for consistency I would rather add it here too. Plus for now FE requested it by id.

Reply:
This is fine from my side. Nevertheless, while consistency is good in many cases, sometimes is better to not to follow it if we get a clear benefit. This is one of those cases. I'd personally deprecate all those existing single deletion endpoints as they're legacy, in an controlled way to not to break their users.

This applies in general also for single creation endpoints etc.

Reply:
Removed deleted by id